txcache/accountBreadcrumb.go

```diff
@@ -1,9 +1,22 @@
 package txcache
 
-import "math/big"
+import (
+	"math/big"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+)
 
 type accountBreadcrumb struct {
-	initialNonce    uint64
-	lastNonce       uint64
+	initialNonce    core.OptionalUint64
+	lastNonce       core.OptionalUint64
 	consumedBalance *big.Int
 }
+
+func (breadcrumb *accountBreadcrumb) updateBreadcrumb(transferredValue *big.Int, lastNonce core.OptionalUint64) {
+	if transferredValue != nil {
+		_ = breadcrumb.consumedBalance.Add(breadcrumb.consumedBalance, transferredValue)
```

Comment:
Maybe we should add a constructor such as `newAccountBreadcrumb`, which sets `consumedBalance` to `big.NewInt(0)`, and call this constructor instead of a plain initialization? That is, since a panic would occur here if `breadcrumb.consumedBalance` is nil, then the actual logic to prevent the panic should be attached to this struct, as well.